### PR TITLE
Convert reference pages documentation section to Markdown

### PR DESCRIPTION
**Pull request by lb-**
-  Part of https://github.com/wagtail/wagtail/issues/8383
-  Migrates the rest of the reference/contrib section pages from RST to MD - intentionally two commits to keep the file history
-   [X] Does the code comply with the style guide?
-   [X] For new features: Has the documentation been updated accordingly?

### CODE REVIEW

File: docs/reference/pages/model_reference.md [hunk 1]
```diff
@@ -442,19 +438,18 @@ Methods and properties
     .. automethod:: get_active
 
     .. automethod:: get_display_name
+```
 
+### Translatable Mixin
```
Comment (thibaudcolas) [inline docs/reference/pages/model_reference.md:443]:
```suggestion
## Translatable Mixin
```

File: docs/reference/pages/model_reference.md [hunk 2]
```diff
@@ -442,19 +438,18 @@ Methods and properties
...
-Translatable Mixin
-==================
-
-``TranslatableMixin`` is an abstract model that can be added to any non-page Django model to make it translatable.
+`TranslatableMixin` is an abstract model that can be added to any non-page Django model to make it translatable.
 Pages already include this mixin, so there is no need to add it.
 
-Methods and properties
-~~~~~~~~~~~~~~~~~~~~~~
+#### Methods and properties
```
Comment (thibaudcolas) [inline docs/reference/pages/model_reference.md:448]:
```suggestion
### Methods and properties
```

File: docs/reference/pages/model_reference.md [hunk 3]
```diff
@@ -669,28 +665,27 @@ Database fields
     .. attribute:: password
 
         (text)
+```
 
-``Orderable`` (abstract)
-========================
+##`Orderable` (abstract)
```
Comment (thibaudcolas) [inline docs/reference/pages/model_reference.md:670]:
```suggestion
## `Orderable` (abstract)
```

File: docs/reference/pages/queryset_reference.md [hunk 4]
```diff
@@ -1,36 +1,31 @@
...
+# Page QuerySet reference
 
-All models that inherit from :class:`~wagtail.models.Page` are given some extra QuerySet methods accessible from their ``.objects`` attribute.
+All models that inherit from {class}`~wagtail.models.Page` are given some extra QuerySet methods accessible from their `.objects` attribute.
 
+## Examples
 
-Examples
-========
+### Selecting only live pages
```
Comment (thibaudcolas) [inline docs/reference/pages/queryset_reference.md:7]: Nice to have those as headings!

File: docs/reference/pages/model_recipes.md [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+            return super().serve(request)
+```
+
+{meth}`~wagtail.models.Page.serve` takes a Django request object and returns a Django response object. Wagtail returns a `TemplateResponse` object with the template and context which it generates, which allows middleware to function as intended, so keep in mind that a simpler response object like a `HttpResponse` will not receive these benefits.
+
+With this strategy, you could use Django or Python utilities to render your model in JSON or XML or any other format you'd like.
+
+(overriding_route_method)=
+
+### Adding Endpoints with Custom {meth}`~wagtail.models.Page.route` Methods
```
Comment (thibaudcolas) [inline docs/reference/pages/model_recipes.md:40]: Nice, this does make more sense to me as a sub-heading. And even more so for the tagging ones below.

File: docs/reference/pages/model_recipes.md [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+
+```{note}
+A much simpler way of adding more endpoints to pages is provided by the :mod:`~wagtail.contrib.routable_page` module.
```
Comment (thibaudcolas) [inline docs/reference/pages/model_recipes.md:43]:
```suggestion
A much simpler way of adding more endpoints to pages is provided by the :py:mod:`routable_page <wagtail.contrib.routable_page>` module.
```

File: docs/reference/pages/model_recipes.md [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+
+```html+django
+{% for tag in page.tags.all %}
+    <a href="{% pageurl page.blog_index %}?tag={{ tag }}">{{ tag }}</a>
+{% endfor %}
+```
+
+Iterating through `page.tags.all` will display each tag associated with `page`, while the links back to the index make use of the filter option added to the `BlogIndexPage` model. A Django query could also use the `tagged_items` related name field to get `BlogPage` objects associated with a tag.
+
+The same approach can be used to add tagging to non-page models managed through :ref:`snippets` and :doc:`/reference/contrib/modeladmin/index`. In this case, the model must inherit from `modelcluster.models.ClusterableModel` to be compatible with `ClusterTaggableManager`.
```
Comment (thibaudcolas) [inline docs/reference/pages/model_recipes.md:197]: Not entirely sure why those ones aren’t working.

```suggestion
The same approach can be used to add tagging to non-page models managed through [](snippets) and :doc:`../contrib/modeladmin/index`. In this case, the model must inherit from `modelcluster.models.ClusterableModel` to be compatible with `ClusterTaggableManager`.
```

**Review by thibaudcolas — CHANGES_REQUESTED** [review]

**Review by thibaudcolas — APPROVED** [review]
Thank you @lb-! I took the liberty to fix some of the formatting issues myself, I hope that’s ok.

**Comment by lb-** [comment]
Thanks @thibaudcolas  - really appreciate the cross-check

**Pull request merged**